scipy/odr/__odrpack.c

```diff
@@ -536,21 +536,11 @@ PyObject *odr(PyObject * self, PyObject * args, PyObject * kwds)
     {
       PYERR(PyExc_TypeError, "initbeta must be a sequence");
     }
-  if (!PySequence_Check(py))
-    {
-      /* Checking whether py is an int
-       *
-       * XXX: PyInt_Check for np.int32 instances does not work on python 2.6 -
-       * we should fix this in numpy, workaround by trying to cast to an int
-       * for now */
-      long val;
-
-      PyErr_Clear();
-      val = PyInt_AsLong(py);
-      if (val == -1 && PyErr_Occurred()) {
-        PYERR(PyExc_TypeError,
-              "y must be a sequence or integer (if model is implicit)");
-      }
+//  if (!PySequence_Check(py) && !PyLong_Check(py))
```

Comment:
This seems the more reasonable check but fails because py is np.int64,
I'm not sure what the right check is, but PyNumber_Check seemed reasonable.